system/scripts/ais/basic_monster.cs

```diff
@@ -49,8 +49,16 @@ protected IEnumerable Attack()
 	{
 		SetRunning(true);
 
+		var master = GetMaster();
+
 		while (!target.IsDead)
-		{
+		{			
+			if (master != null)
+			{
+				if (!MasterAttackCheck(master))
```

Comment:
Is this the appropriate place to do this or should it be an event like the other checks at the bottom of the script? Seems to me like it would be more consistent to handle all "target loss events" in the same way.

Reply:
I moved it to an event.